lms/djangoapps/courseware/tests/test_credit_requirements.py

```diff
@@ -133,6 +136,34 @@ def test_credit_requirements_not_eligible(self):
         )
         self.assertContains(response, "Verification Failed")
 
+    @ddt.data(
+        (CourseMode.VERIFIED, True),
+        (CourseMode.CREDIT_MODE, True),
+        (CourseMode.HONOR, False),
+        (CourseMode.AUDIT, False),
+        (CourseMode.PROFESSIONAL, False),
+        (CourseMode.NO_ID_PROFESSIONAL_MODE, False)
+    )
+    @ddt.unpack
+    def test_credit_requirements_on_progress_page(self, enrollment_mode, display_requirement):
+        # Test the progress table is only displayed to the
+        # verified and credit students.
+        self.enrollment.mode = enrollment_mode
+        self.enrollment.save()  # pylint: disable=no-member
+
+        # Check the progress page display
+        response = self._get_progress_page()
+        if display_requirement:
+            # Assert that the credit-eligibility section is present in the response.
+            self.assertContains(response, "credit-eligibility")
+            # Assert that the eligibility heading div is present in the response.
+            self.assertContains(response, "eligibility-heading")
+        else:
+            # Assert that the credit-eligibility section is not present in the response.
+            self.assertNotContains(response, "credit-eligibility")
```

Comment:
@rlucioni Is this the right way to check the `"credit-eligibility"` section not present in the response?